### PR TITLE
Feedback edu

### PR DESCRIPTION
**Pull request by sigurdthecoder**
## Description
Multiple backend changes. 

## Changes
We added a script for training the AI. 
We have given the AI chatbot access to the list of courses and now it can help tutor users with questions and exercises about the courses. 
We made the backend logic for users to give a thumbs up or thumbs down in the chat to a response, and also for the processing of the thumbs up or thumbs down in the backend. 
We wrote unit tests for AiRoutes. 
Lastly, we made it possible for chatbot responses to be read aloud on the app. 

## Related Issues
No related issues. 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.
- [x] Changes have been approved by Product Owner
- [x] Code has been analyzed with CodeScene
- [x] I have reviewed the pull request myself

## Screenshots (if applicable)

No screenshots, it's all backend changes. 

Commit id: 032536d

## Notes for Reviewers
We love you Anders <3

### CODE REVIEW

**Comment by sigurdthecoder** [comment]
All checks passed and ready to review and merge ;)
We already merged dev into it before creating the PR, so there should be no problemos.

File: package.json [hunk 1]
```diff
@@ -65,9 +65,9 @@
     "@types/jest": "^29.5.14",
     "babel-jest": "^24.4.0",
     "cross-env": "^7.0.3",
-    "jest": "^29.1.2",
+    "jest": "^29.7.0",
```
Comment (Andersraae) [inline package.json:68]: Should these be updated?

File: routes/aiRoutes.js [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+		res.json(feedbacks);
+	} catch (error){
+		console.error('error fetching feedbacks', error.message);
+		res.status(500).json({error: 'Internal server error'});
+	}
+});
+
+router.get('/feedback', shorttermLimiter, longtermLimiter, async (req, res) => {
+    console.log('GET request received at /api/ai/feedback');
+	res.send('feedback route is working!!!???!');
```
Comment (Andersraae) [inline routes/aiRoutes.js:23]: :O

**Review by Andersraae — COMMENTED** [review]
Same as in mobile with the console logs otherwise it looks fine I think

File: routes/aiRoutes.js (hunk 2 above)
Comment (Jenspetersen99) [reply routes/aiRoutes.js:23]: den her er der for testing af endpoints, men kan godt slettes hvis du synes det er dumt at have det der

File: package.json (hunk 1 above)
Comment (Jenspetersen99) [reply package.json:68]: godt spørgsmål, undersøger det lige

**Review by MagnusHoese — APPROVED** [review]
Very impressive implementation! :pinched_fingers:
Thank you for joining me on Discord, where we figured out the rewrite of the files that I told you about. And I'm glad that we figured out that one error! Now it's good to go!

**Pull request merged**